cmd/hc/main.go

```diff
@@ -65,8 +68,19 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	identityPath := filepath.Join(setupCfg.BasePath, "identity")
-	_, err = peertls.NewTLSFileOptions(identityPath, identityPath, true, false)
+	// GenerateCA CA
+	ca := provider.GenerateCA(context.Background(), setupCfg.CA.Difficulty, 4)
+	fi, err := ca.GenerateIdentity()
+	if err != nil {
+		return err
+	}
+
+	err = setupCfg.CA.Save(ca)
```

Comment:
this is surprising to me. what was basepath for?

should ca have a save method that takes a path?